packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUnifyingIndexHtmlRequestListener.java

```diff
@@ -142,18 +140,21 @@ protected Map<String, AvailableViewInfo> collectClientViews(
 
         var clientViews = new HashMap<String, AvailableViewInfo>();
         clientRouteRegistry.getAllRoutes().entrySet().stream()
-                .filter(clientViewConfigEntry -> routeUtil.isRouteAllowed(
-                        isUserInRole, isUserAuthenticated,
-                        clientViewConfigEntry.getValue()))
-                .forEach(clientViewConfigEntry -> {
+                .filter(clientViewConfigEntry -> {
+                    var config = clientViewConfigEntry.getValue();
+                    return routeUtil.isRouteAllowed(isUserInRole,
+                            isUserAuthenticated, config)
+                            && (config.getRouteParameters() == null
+                                    || config.getRouteParameters().isEmpty());
```

Comment:
According to [this](https://github.com/vaadin/hilla/issues/2378#issuecomment-2096002474) comment, the `(config.getRouteParameters() == null || config.getRouteParameters().isEmpty())` which exclude views with parameters (either optional or required) should change to only exclude the routes with required parameters.